### PR TITLE
Fix DCGM Monitor IndexError

### PR DESCRIPTION
**Pull request by Tabrizian**

### CODE REVIEW

File: model_analyzer/monitor/dcgm/dcgm_monitor.py [hunk 1]
```diff
@@ -99,17 +99,13 @@ def _collect_records(self):
...
             # dictionary length
             if len(list(metrics)) > 0:
-                first_key = list(metrics)[0]
-                num_metrics = len(metrics[first_key].values)
-                for i in range(num_metrics):
-                    for tag in self._tags:
-                        dcgm_field = self.model_analyzer_to_dcgm_field[tag]
+                for tag in self._tags:
+                    dcgm_field = self.model_analyzer_to_dcgm_field[tag]
+                    for measurment in metrics[dcgm_field].values:
```
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/dcgm_monitor.py:104]: Should be `measurement`.

File: model_analyzer/monitor/dcgm/dcgm_monitor.py [hunk 2]
```diff
@@ -99,17 +99,13 @@ def _collect_records(self):
...
 
                         # DCGM timestamp is in nanoseconds
                         records.append(
-                            tag(gpu,
-                                float(metrics[dcgm_field].values[i].value),
-                                metrics[dcgm_field].values[i].ts))
+                            tag(gpu, float(measurment.value), measurment.ts))
```
Comment (xprotobeast2) [inline model_analyzer/monitor/dcgm/dcgm_monitor.py:108]: Typo here as well

**Review by xprotobeast2 — CHANGES_REQUESTED** [review]

**Review by xprotobeast2 — APPROVED** [review]

**Comment by dzier** [comment]
What was the issue, and what did this fix?

**Comment by Tabrizian** [comment]
There were some intermittent failures for IndexError in the CI when accessing different metrics. I was not able to replicate it but this will fix any IndexError that will happen.

**Pull request merged**